components/profile-controller/pkg/controller/profile/profile_controller.go

```diff
@@ -139,7 +141,13 @@ func (r *ReconcileProfile) Reconcile(request reconcile.Request) (reconcile.Resul
 		return reconcile.Result{}, err
 	}
 	// No need to update namespace
+	val, ok := foundNs.Annotations["owner"]
+	if (!ok) || val != instance.Spec.Owner.Name {
```

Comment:
I think this if condition is confusing. When will ok to false?
Should we do a else from above so we know the namespace exist?

Also, I am not sure it's the right behavior to return a reconcile error.
I think it will just come back again later forever.
Should we instead set the status of the profile to be invalid or something? @jlewi WDYT?

Reply:
```ok == false``` if user try to create a profile pointing to an existing namespace like default.
Sure we can save status to be failed with the error message.

Reply:
If the namespace is not found, foundNs will be empty and ok also == false right?

I think it's more clear to do a else from line 142:

```
} else { // line 142
  // err != nil, namespace exist
  if foundNs.Annotations["owner"] != instance.Spec.Owner.Name {
    ....
  }
}
```

Reply:
updated